### DISCUSSION TITLE
iperf1.7 in windows-64bit

### DISCUSSION THREAD

**Discussion by hzywhite** [esnet/iperf#1693] [Q&A]
I need to use multicast packet stuffing in the Windows system, so I used iperf1.7-win32, but I always feel that it has some problems, such as negative packet loss rate when out of order occurs, and why does out of order occur.
![IMG_20240504_204000](https://github.com/esnet/iperf/assets/53216231/463a3605-acca-4a70-a9b2-14384ec36164)

**Comment by davidBar-On** [comment]
UDP packets reorder may happen in different network elements, since there is no guaranty for keeping the order. Therefore buffered packets may be sent out of order, by switches, routers, etc.  Another reason is that packets may go through different paths.

In your case, note the relatively large jitter when there are OOO (Out Of Order) packets.  Probably because of some reason in the network path between the client and server the packets are delayed/buffered, and that causes later sending them OOO.  From the displayed output and from looking at the [iperf1.7 code](https://chromium.googlesource.com/external/sctp-refimpl/+/refs/heads/master/APPS/iperf-1.7.0/src/), it seems that no packet is lost. They are just received in different order.

One thing I am not sure about is the report for interval 23-24, were 1060 packets where received, but there where 1880 OOO packets.  The number of packets received is calculated as the [largest message ID received minus the largest ID in the previous interval](https://chromium.googlesource.com/external/sctp-refimpl/+/refs/heads/master/APPS/iperf-1.7.0/src/PerfSocket.cpp#237).  It may be that  duplication of some messages were received.  Duplicate packets can be received when there are problems in the network.  To really find out what happens, network monitors like Wireshark are needed.

By the way, note that this is **iperf3** discussion forum, and that the iperf1.7 you are using (the original iperf tool?) seem to be more than 20 years old.  You may consider swithing to a newer iperf tool, such as iperf3.  iperf3 executable for Windows can be downloaded from [here](https://files.budman.pw/) (not an official iperf3 site, latest version is 3.16).
